universal-login-web3/src/ULWeb3Provider.ts

```diff
@@ -40,9 +40,11 @@ export class ULWeb3Provider implements Provider {
   private readonly sdk: UniversalLoginSDK;
   private readonly walletService: WalletService;
   private readonly uiController: UIController;
+  private unsubscribeNotifications: Nullable<Function> = null;
 
   readonly isLoggedIn: Property<boolean>;
   readonly isUiVisible: Property<boolean>;
+  readonly isNewNotifications: State<boolean> = new State(false);
```

Comment:
I think the code would be better if this property would be a part of `DeployedWallet`'s interface. So something like:
```typescript
class DeployedWallet {
...
  private readonly pendingAuthorizationsState = new State<Notification[]>()
  public  readonly pendingAuthorizations = pendingAuthorizationsState.pipe(
    withEffect(() => this.subscribeAuthorizations(x => this.pendingAuthorizationsState.set(x))),
  )
...
}
```

Than `isNewNotifications` can be declaratively expressed as:
```typescript
readonly isNewNotifications = this.walletService.walletState.pipe(
  map(walletState => {
    if(walletState.kind === 'Deployed') {
       return walletState.deployedWallet.pendingAuthorizations.pipe(
             map(pendingAuthorizations => pendingAuthorizations.length > 0)
       )
     } else {
       return new State(false) // we cannot just return `false` here since the other brach of the `if` statement has the type `Property<boolean>`
    }
  }),
  flat,  // the `map` function above returned `Property<Property<boolean>>` which we can reduce to `Property<boolean>` using the `flat` operator
)
```

Reply:
i think we don't want by default subscribe AuthorisationObserver. If we would it will be next PR

Reply:
The code above will only subscribe to authorisations when somebody starts observing the property. I think it would be better to do it in this PR so not to introduce the complex subscription handling logic in `ULWeb3Provider` which is already has a lot of stuff in it.